importer/api/__init__.py

```diff
@@ -0,0 +1,100 @@
+"""
+Import OLX data into LORE.
+"""
+
+# stdlib
+from glob import glob
+from shutil import rmtree
+from tempfile import mkdtemp
+from os.path import join
+import tarfile
+import zipfile
+
+# PyPi
+
+from xbundle import XBundle, DESCRIPTOR_TAGS
+from lxml import etree
+
+# MITODL
+# commented out for now, until it's merged
+# from lore.learningobjects.models import Course, LearningObject
+
+def import_course_from_file(filename):
+    """
+    Import OLX from .zip or tar.gz.
+    """
+    tempdir = mkdtemp()
+    if filename.endswith(".tar.gz"):
+        course = tarfile.open(filename, "r")
+    elif filename.endswith(".zip"):
+        course = zipfile.ZipFile(filename, "r")
+    else:
+        raise ValueError("Unexpected file type (want tarball or zip).")
+    course.extractall(tempdir)
+    dirs = glob(join(tempdir, "*"))
+    assert len(dirs) == 1
+    import_course_from_path(dirs[0])
+    rmtree(tempdir)
+
+def import_course_from_path(path):
+    """
+    Import course from an OLX directory.
+    """
+    bundle = XBundle()
+    bundle.import_from_directory(path)
+    return import_course(bundle)
+
+
+def import_course(bundle):
+    """
+    Import a course from an XBundle object.
+    """
+    course = Course()
+    course.course_number = bundle.course.attrib.get("course", "MISSING")
+    course.save()
```

Comment:
This means we are overwriting courses with the same ID right? I thought we decided this would be one time only, so we should raise if it already exists I think